ide/src/main/java/com/devonfw/tools/ide/commandlet/Commandlet.java

```diff
@@ -1,70 +1,164 @@
-  private boolean batch;
+  private final List<Property<?>> valuesList;
 
-  @CommandLine.Option(names = { "-f", "--force" }, description = "enable force mode")
-  private boolean force;
+  private final List<Property<?>> values;
 
-  @CommandLine.Option(names = { "-q", "--quiet" }, description = "enable quiet mode (mimimum logging)")
-  private boolean quiet;
+  private final Map<String, Property<?>> optionMap;
 
-  @CommandLine.Option(names = { "-o", "--offline" }, description = "enable offline mode")
-  private boolean offline;
+  private Property<?> multiValued;
 
-  @Override
-  public Integer call() {
+  /**
+   * The constructor.
+   *
+   * @param context the {@link IdeContext}.
+   */
+  public Commandlet(IdeContext context) {
+
+    super();
+    this.context = context;
+    this.propertiesList = new ArrayList<>();
+    this.properties = Collections.unmodifiableList(this.propertiesList);
+    this.valuesList = new ArrayList<>();
+    this.values = Collections.unmodifiableList(this.valuesList);
+    this.optionMap = new HashMap<>();
+  }
+
+  /**
+   * @return the {@link List} with all {@link Property properties} of this {@link Commandlet}.
+   */
+  public List<Property<?>> getProperties() {
 
-    IdeLogLevel logLevel = IdeLogLevel.INFO;
-    if (this.debug) {
-      logLevel = IdeLogLevel.DEBUG;
+    return this.properties;
+  }
+
+  /**
+   * @return the {@link List} of {@link Property properties} that are {@link Property#isValue() values}.
+   */
+  public List<Property<?>> getValues() {
+
+    return this.values;
+  }
+
+  /**
+   * @param nameOrAlias the potential {@link Property#getName() name} or {@link Property#getAlias() alias} of the
+   *        requested {@link Property}.
+   * @return the requested {@link Property property} or {@code null} if not found.
+   */
+  public Property<?> getOption(String nameOrAlias) {
+
+    return this.optionMap.get(nameOrAlias);
+  }
+
+  /**
+   * @param keyword the {@link KeywordProperty keyword} to {@link #add(Property) add}.
+   */
+  protected void addKeyword(String keyword) {
+
+    add(new KeywordProperty(keyword, true, null));
+  }
+
+  /**
+   * @param <P> type of the {@link Property}.
+   * @param property the {@link Property} to register.
+   * @return the given {@link Property}.
+   */
+  protected <P extends Property<?>> P add(P property) {
+
+    if (this.multiValued != null) {
+      throw new IllegalStateException(
+          "The multi-valued property " + this.multiValued + " can not be followed by " + property);
+    }
+    this.propertiesList.add(property);
+    if (property.isOption()) {
+      add(property.getName(), property, false);
+      add(property.getAlias(), property, true);
+    }
+    if (property.isValue()) {
+      this.valuesList.add(property);
+    }
+    if (property.isMultiValued()) {
+      this.multiValued = property;
+    }
+    return property;
+  }
+
+  private void add(String name, Property<?> property, boolean alias) {
```

Comment:
This is clearly a point where I would strongly disagree.
There is a protected `add` method what is the API for commandlets inheriting from this class.
It does have JavaDoc. This private method only exists to avoid code redundancy and the only place where it is called is the protected `add` method. So IMHO a user would only be interested in the API and if he digs in the implementation of the protected `add` method and will stumble over the private one he can easily read the 12 lines of code that are purely trivial.
I would rather challenge that the existing JavaDoc like `@param property the {@link Property} to register.` might be a little bit short and could explain what "registration" means here. Ideally we will live a culture where every developer takes responsibility of the code and feels invited to improve JavaDoc, add JUnit coverage, etc. to all existing code.
I try my best to produce high quality code but only have very few hours a day for management, contributing and coding to this project. So sure I could write more JavaDoc, more JUnits (that is IMHO even a much bigger pain point currently).
Do not get me wrong:
Thank you for your review and feedback. I just wanted to make a clear point to match expectations.

Reply:
Thanks for pointing that out. I'm asking for javadocs only so that I don't have to read the method body multiple times as mentioned before. Aren't there tools which could automate the javadoc and param description using AI yet?